code/modules/economy/holopay.dm

```diff
@@ -0,0 +1,249 @@
+			balloon_alert(user, "invalid account")
+			return FALSE
+		/// Delete the holopay if the master swipes on it
+		if(pay_card.registered_account == linked_account)
+			qdel(src)
+			return TRUE
+		process_payment(user)
+		return TRUE
+	/// Users can also pay by holochip
+	if(istype(held_item, /obj/item/holochip))
+		/// Account checks
+		var/obj/item/holochip/chip = held_item
+		if(!chip.credits)
+			balloon_alert(user, "holochip is empty")
+			to_chat(user, span_warning("There doesn't seem to be any credits here."))
+			return FALSE
+		if(force_fee && chip.credits < force_fee)
+			balloon_alert(user, "insufficient credits")
+			to_chat(user, span_warning("You don't have enough credits to pay with this chip."))
+			return FALSE
+		/// Charges force fee or uses pay what you want
+		var/cash_deposit = force_fee || tgui_input_number(user, "How much? (Max: [chip.credits])", "Patronage", max_value = chip.credits)
+		/// Exit sanity checks
+		if(!cash_deposit)
+			return TRUE
+		if(QDELETED(held_item) || QDELETED(user) || QDELETED(src) || !user.canUseTopic(src, BE_CLOSE, FALSE, NO_TK))
+			return FALSE
+		if(!chip.spend(cash_deposit, FALSE))
+			balloon_alert(user, "insufficient credits")
+			to_chat(user, span_warning("You don't have enough credits to pay with this chip."))
+			return FALSE
+		/// Success: Alert buyer
+		alert_buyer(user, cash_deposit)
+		return TRUE
+
+	/// Throws errors if they try to use space cash
+	if(istype(held_item, /obj/item/stack/spacecash))
+		to_chat(user, "What is this, the 2000s? We only take card here.")
+		return TRUE
+	if(istype(held_item, /obj/item/coin))
+		to_chat(user, "What is this, the 1800s? We only take card here.")
+		return TRUE
+	return ..()
+
+/**
+ * Initiates a transaction between accounts.
+ *
+ * Parameters:
+ * * user - The user who initiated the transaction.
+ * Returns:
+ * * TRUE if the transaction was successful, FALSE otherwise.
+ */
+/obj/machinery/holopay/proc/process_payment(mob/living/user)
+	// Preliminary sanity checks
+	if(!isliving(user) || issilicon(user))
+		return FALSE
+	/// Account checks
+	var/obj/item/card/id/id_card
+	id_card = user.get_idcard(TRUE)
+	if(!id_card || !id_card.registered_account || !id_card.registered_account.account_job)
+		balloon_alert(user, "invalid account")
+		to_chat(user, span_warning("You don't have a valid account."))
+		return FALSE
+	var/datum/bank_account/payee = id_card.registered_account
+	if(payee == linked_account)
+		balloon_alert(user, "invalid transaction")
+		to_chat(user, span_warning("You can't pay yourself."))
+		return FALSE
+	if(!payee.has_money(force_fee || 1))
+		balloon_alert(user, "insufficient credits")
+		to_chat(user, span_warning("You don't have the money to pay for this."))
+		return FALSE
+	/// If the user has enough money, ask them the amount or charge the force fee
+	var/amount = force_fee || tgui_input_number(user, "How much? (Max: [payee.account_balance])", "Patronage", max_value = payee.account_balance)
```

Comment:
Why do the balance check before you get the amount they want to pay?

Reply:
This is only in the case they try to pay but don't have the force fee or any money at all

Reply:
aren't you checking for not having enough twice here? Is the goal to prevent people with no money from buying something?